component/otelcol/processor/probabilistic_sampler/probabilistic_sampler_test.go

```diff
@@ -0,0 +1,251 @@
+//go:build !race
+
+package probabilistic_sampler
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/go-kit/log/level"
+	"github.com/grafana/agent/component/otelcol/internal/fakeconsumer"
+	"github.com/grafana/agent/pkg/util"
+
+	"github.com/grafana/agent/component/otelcol"
+	"github.com/grafana/agent/pkg/flow/componenttest"
+	"github.com/grafana/dskit/backoff"
+	"github.com/grafana/river"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+func TestBadRiverConfigNegativeSamplingRate(t *testing.T) {
+	exampleBadRiverConfig := `
+    sampling_percentage = -1
+    output { 
+	    // no-op: will be overridden by test code.
+    }
+`
+	var args Arguments
+	require.EqualError(t, river.Unmarshal([]byte(exampleBadRiverConfig), &args), "negative sampling rate: -1.00")
+}
+
+func TestBadRiverConfigInvalidAttributeSource(t *testing.T) {
+	exampleBadRiverConfig := `
+    sampling_percentage = 0.1
+    attribute_source = "example"
+    output { 
+	    // no-op: will be overridden by test code.
+    }
+`
+	var args Arguments
+	require.EqualError(t, river.Unmarshal([]byte(exampleBadRiverConfig), &args), "invalid attribute source: example. Expected: traceID or record")
+}
+
+func TestLogProcessing(t *testing.T) {
```

Comment:
Those tests have a lot of boilerplate. It would be cleaner to do something like those ones:
https://github.com/grafana/agent/blob/main/component/otelcol/connector/spanlogs/spanlogs_test.go
https://github.com/grafana/agent/blob/main/component/otelcol/processor/attributes/attributes_test.go